MediaGalleryUi/Model/AssetIndexer.php

```diff
@@ -71,13 +71,16 @@ public function __construct(
     public function execute(\SplFileInfo $item): void
     {
         $file = $item->getPath() . '/' . $item->getFileName();
+        $relativePath = $this->getMediaDirectory()->getRelativePath($file);
 
+        $path = dirname($relativePath) === '.' ? '/' . $relativePath : $relativePath;
```

Comment:
Please extract path calculation to the `private function getPath(\SplFileInfo $item): string`